modular/blocksyncer/blocksyncer_options.go

```diff
@@ -1,478 +0,0 @@
-			RPC: &remote.RPCConfig{
-				ClientName: "juno",
-				Address:    rpcAddress,
-			},
-		},
-		Parser: parserconfig.Config{
-			Workers: int64(cfg.BlockSyncer.Workers),
-		},
-		Database: databaseconfig.Config{
-			Type:               "mysql",
-			DSN:                cfg.BlockSyncer.Dsn,
-			PartitionBatchSize: 10_000,
-			MaxIdleConnections: 10,
-			MaxOpenConnections: 30,
-		},
-		Logging: loggingconfig.Config{
-			Level: "debug",
-		},
-		EnableDualDB: cfg.BlockSyncer.EnableDualDB,
-		DsnSwitched:  cfg.BlockSyncer.DsnSwitched,
-	}
-}
-
-func newBackupBlockSyncerService(cfg *config.TomlConfig, mainDBIsMaster bool) (*BlockSyncerModular, error) {
-	backUpConfig, err := generateConfigForBackup(cfg)
-	if err != nil {
-		return nil, err
-	}
-
-	BackupService = &BlockSyncerModular{
-		config: backUpConfig,
-		name:   BlockSyncerModularBackupName,
-	}
-
-	if err = BackupService.initClient(); err != nil {
-		return nil, err
-	}
-
-	// init meta db, if mainService db is not current master, backup is master, don't recreate
-	if err = BackupService.initDB(false); err != nil {
-		return nil, err
-	}
-	return BackupService, nil
-}
-
-func generateConfigForBackup(cfg *config.TomlConfig) (*config.TomlConfig, error) {
-	configBackup := new(config.TomlConfig)
-	if err := DeepCopyByGob(cfg, configBackup); err != nil {
-		return nil, err
-	}
-
-	configBackup.Database.DSN = configBackup.DsnSwitched
-
-	return configBackup, nil
-}
-
-func DeepCopyByGob(src, dst interface{}) error {
-	var buffer bytes.Buffer
-	if err := gob.NewEncoder(&buffer).Encode(src); err != nil {
-		return err
-	}
-	return gob.NewDecoder(&buffer).Decode(dst)
-}
-
-func getDBConfigFromEnv(dsn string) (string, error) {
-	dsnVal, ok := os.LookupEnv(dsn)
-	if !ok {
-		return "", ErrDSNNotSet
-	}
-	return dsnVal, nil
-}
-
-func Cast(indexer parser.Indexer) *Impl {
-	s, ok := indexer.(*Impl)
-	if !ok {
-		panic("cannot cast")
-	}
-	return s
-}
-
-func CheckProgress() {
-	for {
-		epochMaster, err := MainService.parserCtx.Database.GetEpoch(context.TODO())
-		if err != nil {
-			continue
-		}
-		epochSlave, err := BackupService.parserCtx.Database.GetEpoch(context.TODO())
-		if err != nil {
-			continue
-		}
-		if epochMaster.BlockHeight-epochSlave.BlockHeight < DefaultBlockHeightDiff {
-			SwitchMasterDBFlag()
-			StopMainService()
-			break
-		}
-		time.Sleep(time.Minute * DefaultCheckDiffPeriod)
-	}
-}
-
-func SwitchMasterDBFlag() error {
-	masterFlag, err := FlagDB.GetMasterDB(context.TODO())
-	if err != nil {
-		return err
-	}
-
-	//switch flag
-	masterFlag.IsMaster = !masterFlag.IsMaster
-	if err = FlagDB.SetMasterDB(context.TODO(), masterFlag); err != nil {
-		return err
-	}
-	log.Infof("DB switched")
-	return nil
-}
-
-func StopMainService() error {
-	CancelMain()
-	return nil
-}
-
-// mustGetLatestHeight tries getting the latest height from the RPC client.
-// If no latest height can be found after MaxRetryCount, it returns 0.
-func mustGetLatestHeight(ctx *parser.Context) uint64 {
```

Comment:
`mustGetLatestHeight`, to me the `must` means if it can not fetch the height in time, it should panic.

Reply:
Good advice. I'll change it